tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs

```diff
@@ -666,6 +665,33 @@ public void TiffDecoder_CanDecode_Fax4CompressedWithStrips<TPixel>(TestImageProv
     public void TiffDecoder_CanDecode_TiledWithNonEqualWidthAndHeight<TPixel>(TestImageProvider<TPixel> provider)
         where TPixel : unmanaged, IPixel<TPixel> => TestTiffDecoder(provider);
 
+    [Theory]
+    [WithFile(JpegCompressedGray0000539558, PixelTypes.Rgba32)]
+    public void TiffDecoder_ThrowsException_WithCircular_IFD_Offsets<TPixel>(TestImageProvider<TPixel> provider)
+        where TPixel : unmanaged, IPixel<TPixel>
+        => Assert.Throws<ImageFormatException>(
+            () =>
+            {
+                using (provider.GetImage(TiffDecoder.Instance))
+                {
+                }
+            });
+
+    [Theory]
+    [WithFile(Tiled0000023664, PixelTypes.Rgba32)]
+    public void TiffDecoder_CanDecode_TiledWithBadZlib<TPixel>(TestImageProvider<TPixel> provider)
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        using Image<TPixel> image = provider.GetImage(TiffDecoder.Instance);
+
+        // ImageMagick cannot decode this image.
+        image.DebugSave(provider);
+        image.CompareToReferenceOutput(
+            ImageComparer.Exact,
```

Comment:
Isn't there uninialized memory garbage after the exit criteria has been reached? If so, this test might be flaky. Or is there some logic in tiff that guarantees those scanlines it to be black in tiff?

(If not, clearing it out would be beyond the scope of this fix.)

Reply:
Each frame is decoded into a clean buffer.

<img alt="image" src="https://github.com/SixLabors/ImageSharp/assets/385879/62407f4a-de57-4478-8a85-90c81f8904a8">

Reply:
I would prefer it to somehow create frames with unintialized buffers, but that's a topic for another day.

Reply:
Me too but I didn’t want to risk greater changes. There’s too much going on in the Tiff decoder I don’t fully understand